### DISCUSSION TITLE
Corrupted h264 stream from rtsp webcam

### DISCUSSION THREAD

**Discussion by marcin-koziol** [bluenviron/mediamtx#356] [Questions]
## Which version are you using?

v0.15.2

## Which operating system are you using?

OS

- [ X] Linux
- [ ] Windows
- [ ] macOS

Architecture

- [ X] amd64
- [ ] arm64
- [ ] arm7
- [ ] arm6

## Describe the issue
I'm trying to re-stream from webcam, using rtsp, then play the stream using vlc/mpv, using simple config (I've set "path" to "test" and source to my camera's RTSP URL

## Describe how to replicate the issue
1. mpv CAMERA_URL - works fine
![Zrzut ekranu z 2021-03-31 14-09-12](https://user-images.githubusercontent.com/1050057/113142977-e0efe800-922b-11eb-93c5-8a7259fd4028.png)

2. run the server with default config, replaced path and source:
```
paths:
  test:
    source: rtsp://login:pass@192.168.21.37:554/stream2
```

3. play the server's stream using mpv:
![Zrzut ekranu z 2021-03-31 14-08-52](https://user-images.githubusercontent.com/1050057/113143032-f107c780-922b-11eb-8a9e-3dcaaee7b801.png)
```
 mpv rtsp://localhost:8554/test
[ffmpeg/video] h264: left block unavailable for requested intra mode
[ffmpeg/video] h264: error while decoding MB 0 2, bytestream 1544
 (+) Video --vid=1 (h264 640x360 15.000fps)
 (+) Audio --aid=1 (pcm_alaw 1ch 8000Hz)
File tags:
 Title: Stream
[ffmpeg/video] h264: left block unavailable for requested intra mode
[ffmpeg/video] h264: error while decoding MB 0 2, bytestream 1544
AO: [pulse] 8000Hz mono 1ch s16
VO: [gpu] 640x360 yuv420p
AV: 00:00:02 / 00:00:04 (57%) A-V:  0.044
[ffmpeg/video] h264: left block unavailable for requested intra mode
[ffmpeg/video] h264: error while decoding MB 0 3, bytestream 1128
AV: 00:00:06 / 00:00:08 (78%) A-V: -0.000
[ffmpeg/video] h264: left block unavailable for requested intra mode
[ffmpeg/video] h264: error while decoding MB 0 2, bytestream 1476
AV: 00:00:08 / 00:00:10 (82%) A-V:  0.003
[ffmpeg/video] h264: error while decoding MB 5 16, bytestream -6
AV: 00:00:08 / 00:00:10 (82%) A-V:  0.008
[ffmpeg/video] h264: left block unavailable for requested intra4x4 mode -1
[ffmpeg/video] h264: error while decoding MB 0 2, bytestream 1288
AV: 00:00:10 / 00:00:12 (84%) A-V: -0.002
[ffmpeg/video] h264: error while decoding MB 5 17, bytestream -8
AV: 00:00:10 / 00:00:12 (85%) A-V: -0.001
```

**Comment by aler9** [comment]
Hello,
* please try to increase `readBufferCount` in the configuration to at least `1024`, and see if there's a difference
* Please try to set `sourceProtocol: tcp` and see if there's a difference.

Frame corruption can happen because

* there isn't enough bandwidth between the camera and the server, or between the server and the camera
* the ring buffer of the server is not big enough to handle the frames queue
* UDP used to have problems with high bitrates, but they should be solved, unless there's a regression

**Comment by marcin-koziol** [comment]
Right, it was caused by slow connection.

[a comment by an automated account is not shown]
